StripeConnect/StripeConnect/Source/Internal/Webview/ConnectWebView.swift

```diff
@@ -89,18 +98,50 @@ private extension ConnectWebView {
         return popupVC.webView
     }
 
-    // Opens the given URL in an SFSafariViewController
+    /// Opens the given URL in an SFSafariViewController
     func openInAppSafari(url: URL) {
         let safariVC = SFSafariViewController(url: url)
         safariVC.dismissButtonStyle = .done
         safariVC.modalPresentationStyle = .popover
         presentPopup(safariVC)
     }
 
-    // Opens with UIApplication.open, if supported
+    /// Opens with UIApplication.open, if supported
     func openOnSystem(url: URL) {
         urlOpener.openIfPossible(url)
     }
+
+    /// Returns true if this URL should open in an embedded web view
+    func shouldNavigate(toURL url: URL) -> Bool {
+        // Only open known hosts inside an embedded web view, otherwise use
+        // an SFSafariViewController so the user can have navigation controls
+
+        ["http", "https"].contains(url.scheme)
+        && url.host.map(StripeConnectConstants.allowedHosts.contains) != false
+    }
+
+    /// Returns true if URL is handled externally
+    func openInSystemOrSafari(_ url: URL) {
+        // Only `http` or `https` URL schemes can be opened in WKWebView or
+        // SFSafariViewController. Opening other schemes, like `mailto`, will
+        // cause a fatal error.
+        if !["http", "https"].contains(url.scheme) {
+            openOnSystem(url: url)
+        } else {
+            openInAppSafari(url: url)
+        }
+    }
```

Comment:
DRYed these up from just handling popups so they could be reused when receiving response redirects from not allow-listed domains